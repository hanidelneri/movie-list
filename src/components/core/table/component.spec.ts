import { mount } from "@/tests/utils";
import { Movie } from "@/types";
import Table from "./component.vue";

const mock: Movie[] = [
  {
    Title: "movie-1",
    Year: 1999,
    imdbID: "1"
  },
  {
    Title: "movie-2",
    Year: 1998,
    imdbID: "2"
  }
];

describe("core/table", () => {
  beforeEach(() => {
    localStorage.setItem("favourite-movies", JSON.stringify([]));
  });
  test("add to favourites works correctly", async () => {
    const wrapper = mount(Table, {
      propsData: {
        movies: mock
      }
    });
    const stars = wrapper.findAll("tbody > tr button").wrappers;
    await stars[0].trigger("click");

    let stored = localStorage.getItem("favourite-movies");
    if (!stored) stored = "";
    const movies = JSON.parse(stored) as Movie[];
    expect(movies[0].Title).toBe(mock[0].Title);
    expect(movies[0].Year).toBe(mock[0].Year);
    expect(movies[0].imdbID).toBe(mock[0].imdbID);
    expect(wrapper).toMatchSnapshot();
  });

  test("remove from favourites works correctly", async () => {
    const wrapper = mount(Table, {
      propsData: {
        movies: mock
      }
    });
    const star = wrapper.findAll("tbody > tr button").wrappers[0];
    await star.trigger("click");
    await wrapper.vm.$nextTick();
    const remove = wrapper.findAll("tbody > tr button").wrappers[0];
    await remove.trigger("click");
    await wrapper.vm.$nextTick();
    let stored = localStorage.getItem("favourite-movies");
    if (!stored) stored = "";
    const movies = JSON.parse(stored) as Movie[];
    expect(movies).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });
});
