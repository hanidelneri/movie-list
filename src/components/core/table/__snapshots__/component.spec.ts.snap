// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core/table add to favourites works correctly 1`] = `
<div class="v-data-table elevation-1 theme--light">
  <div class="v-data-table__wrapper">
    <table>
      <colgroup>
        <col class="">
        <col class="">
        <col class="">
        <col class="">
      </colgroup>
      <thead class="v-data-table-header">
        <tr>
          <th role="columnheader" scope="col" aria-label="Favourite" class="text-start"><span>Favourite</span></th>
          <th role="columnheader" scope="col" aria-label="ID" class="text-start"><span>ID</span></th>
          <th role="columnheader" scope="col" aria-label="Year" class="text-start"><span>Year</span></th>
          <th role="columnheader" scope="col" aria-label="Title" class="text-start"><span>Title</span></th>
        </tr>
      </thead>
      <tbody>
        <tr class="">
          <td class="text-start"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-star theme--light yellow--text"></button></td>
          <td class="text-start">1</td>
          <td class="text-start">1999</td>
          <td class="text-start">movie-1</td>
        </tr>
        <tr class="">
          <td class="text-start"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-star-outline theme--light"></button></td>
          <td class="text-start">2</td>
          <td class="text-start">1998</td>
          <td class="text-start">movie-2</td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`core/table remove from favourites works correctly 1`] = `
<div class="v-data-table elevation-1 theme--light">
  <div class="v-data-table__wrapper">
    <table>
      <colgroup>
        <col class="">
        <col class="">
        <col class="">
        <col class="">
      </colgroup>
      <thead class="v-data-table-header">
        <tr>
          <th role="columnheader" scope="col" aria-label="Favourite" class="text-start"><span>Favourite</span></th>
          <th role="columnheader" scope="col" aria-label="ID" class="text-start"><span>ID</span></th>
          <th role="columnheader" scope="col" aria-label="Year" class="text-start"><span>Year</span></th>
          <th role="columnheader" scope="col" aria-label="Title" class="text-start"><span>Title</span></th>
        </tr>
      </thead>
      <tbody>
        <tr class="">
          <td class="text-start"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-star-outline theme--light"></button></td>
          <td class="text-start">1</td>
          <td class="text-start">1999</td>
          <td class="text-start">movie-1</td>
        </tr>
        <tr class="">
          <td class="text-start"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-star-outline theme--light"></button></td>
          <td class="text-start">2</td>
          <td class="text-start">1998</td>
          <td class="text-start">movie-2</td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
